File: Front/.Net Maui/BetFriends.Domain.UnitTests/Implems/MockUserGateway.cs
using BetFriends.Domain.Abstractions;

namespace BetFriends.Domain.UnitTests.Implems;

internal class MockUserGateway : IUserGateway
{
    public User? User { get; internal set; }

    public Task SaveAsync(User user)
    {
        User = user;
        return Task.CompletedTask;
    }
}

File: Front/TS/tests/unit/implems/MockUserGateway.ts
import { IUserGateway } from "../../../src/domain/abstractions/IUserGateway";

export class MockUserGateway implements IUserGateway {
    Register(user: { id: string; username: string; email: string; password: string; }): Promise<void> {
        this.User = user;
        return Promise.resolve();
    }
    User: any;

}
